int32 x
int32 y
float32 vel_x
float32 vel_z